apputils/config_manager.py

```diff
@@ -0,0 +1,83 @@
+"""Configuration Manager."""
+from celery.schedules import crontab
+from config_validator import ConfigValidator
+from string import Template
+import ConfigParser
+import os
+
+
+class ConfigManager(object):
+    def __init__(self, config_path):
+        self.scheduled_tasks = {}
+        self.load_config_files(config_path)
+
+    def load_config_files(self, config_path):
+        """Load all valid config files from path, print task config to stdout.
+
+        Args:
+            config_path(str): Path to configuration directory.
+
+        """
+        files_in_dir = [os.path.join(config_path, f)
+                        for f in os.listdir(config_path)
+                        if os.path.isfile(os.path.join(config_path, f))]
+        for target in sorted(files_in_dir):
+            print("ConfigManager: Parsing config file: %s" % target)
+            config = self.get_scheduled_task_config_from_file(target)
+            if ConfigValidator.validate_config(config):
+                task_name = config["task_config"]["task_name"]
+                self.scheduled_tasks[task_name] = config
+                # Coerce `read_only` to boolean
+                if self.scheduled_tasks[task_name]["task_config"]["read_only"] == "true":  # NOQA
+                    self.scheduled_tasks[task_name]["task_config"]["read_only"] = True  # NOQA
+                else:
+                    self.scheduled_tasks[task_name]["task_config"]["read_only"] = False  # NOQA
+        self.print_tasks(self.scheduled_tasks)
+
+    def beat_tasks_from_config(self):
+        """Return dictionary that describes celerybeat tasks, from config."""
+        beats = {}
+        for j in self.scheduled_tasks.items():
+            beats[j[0]] = {'task': 'halocelery.tasks.generic_bound_containerized_task',  # NOQA
```

Comment:
move this logic to a function. pass in J to a builder function so you can test building the hash correctly from a single scheduled task item. can also assert each item/value is expected

Reply:
Done